### PR TITLE
[Feature] Edit Component - 1130

### LINKED ISSUE #1130: Enable admins to rename a system component (e.g., element)
**Story:**

As a GovReady administrator, 
I want an edit button near the name of a component in the Component Library pages,
so that I can update the name of a component.

**Acceptance criteria:**

GIVEN I have GovReady admin privileges,
WHEN I am on the component detail page for a component in the component library,
THEN I see an edit button to update the name of a component.

GIVEN I have GovReady admin privileges,
and GIVEN I am on the component detail page for a component in the component library,
and GIVEN I see an edit button to update the name of a component,
WHEN I click the component name edit button,
THEN I see simple pop-up to rename the component (or cancel) the action.

GIVEN I have GovReady admin privileges,
and GIVEN I have clicked the component name edit button,
WHEN I enter a new name and save the change,
THEN the name of the component is changed in the database,
AND the name change is logged using the existing logger infrastructure,
AND the name updates on the web page I am viewing.

A very simple initial unit test is written to updating the name of a component.

A very simple initial UI test is written to test the existence of the button on the component page if the user is an admin.

### PR DESCRIPTION
**Pull request by Yodart**
# Pull Request

## Description

Added the ability for a GovReady administrator to rename a Component with the use of an edit button near the name of a component in the Component Library pages.

Closes #1130 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Unit Test - Tests updating the name of a component 
- [X] UI Test - Tests the existence of the button on the component page if the user is an admin
Scripts have been implemented to `controls/tests.py` to run it: `python manage.py test controls.tests `

## Checklist:

- [ ] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [X] I have checked my code and corrected any misspellings

### CODE REVIEW

File: controls/views.py [hunk 1]
```diff
@@ -206,6 +206,29 @@ def controls_updated(request, system_id):
...
+    """Update the component's name
+    Args:
+        request ([HttpRequest]): The network request
+    component_id ([int|str]): The id of the component
+    Returns:
+        [JsonResponse]: Either a ok status or an error 
+    """
+    try:
+        new_name = request.POST.get("name", "").strip() or None
+        element = get_object_or_404(Element, id=element_id)
```
Comment (davidpofo) [inline controls/views.py:219]: not controls_element(Element), edit the siteapp_portfolio table(Portfolio). Which includes description and title.

**Review by davidpofo — COMMENTED** [review]
This PR changes element names but not portfolio titles & descriptions. Closing for rework.

**Comment by davidpofo** [comment]
Needs to change portfolio information not element

**Comment by gregelin** [comment]
@davidpofo There is a need to edit/change a component and change portfolio. These are separate items. I am re-opening PR.

File: controls/views.py (hunk 1 above)
Comment (gregelin) [reply controls/views.py:219]: Editing component and editing portfolio are two different tasks. Both are needed.

**Review by gregelin — APPROVED** [review]
This code all looks good. Tests look good, too.

**Comment by gregelin** [comment]
@Yodart Sorry PR #1157 for issue #1130 took so long to merge.

**Pull request merged**